src/System.Private.CoreLib/src/System/Reflection/Emit/MethodBuilder.cs

```diff
@@ -87,12 +87,12 @@ public sealed class MethodBuilder : MethodInfo
             m_module = mod;
             m_containingType = type;
 
-            // 
-            //if (returnType == null)
-            //{
-            //    m_returnType = typeof(void);
-            //}
-            //else
+
+            if (returnType == null)
+            {
+                m_returnType = typeof(void);
+            }
+            else
```

Comment:
Interesting.  This commented out code appears to have been commented out for a very long time.  Is this just a long-standing bug?

Reply:
Yes, the commented out code was a bug fix that was rolled back for bug-for-bug compatibility.